### ISSUE TITLE
rotation of rigid bodies

### ISSUE THREAD

**Issue by RobinSchmidt** [RobinSchmidt/RS-MET#291]
i just came across this:

https://www.youtube.com/watch?v=1VPfZ_XzisU

...this sort of motion can be described by a simple system of 3 differential equations (pretty much like the lorenz system), see here:

https://en.wikipedia.org/wiki/Tennis_racket_theorem

just for fun, i implemented it and got the following result:

![image](https://user-images.githubusercontent.com/1833099/65393444-e8098080-dd80-11e9-8b4b-927fafd4df58.png)

blue: w2 - the angular velocity about the axis with the intermediate moment of inertia. black and green: the angular velocities about the axis with minimum and maximum moment of inertia respectively. all by itself, it's probably not particularly interesting for musical purposes.  it basically just generates smoothed out square-waves or smoothed unipolar or bipolar impulse trains. but maybe it could be turned into some interesting chaotic system by modifying the ODE system a bit? ...dunno.... i'm just having some fun

**Comment by elanhickler** [comment]
could make a good chaos modulator or noise generator?

**Comment by RobinSchmidt** [comment]
maybe. i just added an extra term to the equations that leads to asymmetric pulses (i.e. pulse-width control for the square wave):

![image](https://user-images.githubusercontent.com/1833099/65393758-1a1ce180-dd85-11e9-93ff-7fd322b201b3.png)

 ...but it also leads to decay (undesirable) but i have ideas how to avoid the decay (by enforcing constant angular momentum and/or rotational energy)

**Comment by RobinSchmidt** [comment]
ha! there's also a version of the equations that includes external torques:
https://en.wikipedia.org/wiki/Euler%27s_equations_(rigid_body_dynamics)
...that means, we can use it to process incoming signals instead of just generating signals on its own

**Comment by elanhickler** [comment]
yes, i was just about to say that. you can make it into a resonator.
